src/measure_theory/l2_space.lean

```diff
@@ -0,0 +1,132 @@
+/-
+Copyright (c) 2021 Rémy Degenne. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Rémy Degenne
+-/
+import analysis.normed_space.inner_product
+import measure_theory.set_integral
+
+/-! # `L^2` space
+
+-/
+
+noncomputable theory
+open topological_space measure_theory measure_theory.Lp
+open_locale nnreal ennreal
+
+namespace measure_theory
+
+variables {α E F 𝕜 : Type*} [is_R_or_C 𝕜] [measurable_space α] {μ : measure α}
+  [measurable_space E] [inner_product_space 𝕜 E] [borel_space E] [second_countable_topology E]
+  [normed_group F] [measurable_space F] [borel_space F] [second_countable_topology F]
+
+local notation `⟪`x`, `y`⟫` := @inner 𝕜 E _ x y
+
+lemma two_mul_le_add_sq (a b : ℝ) : 2 * a * b ≤ a ^ 2 + b ^ 2 :=
```

Comment:
This could move to `analysis/mean_inequalities`. (There are some similar lemmas there, like `nnreal.geom_mean_le_arith_mean2_weighted`, but seemingly not quite this one.)

Reply:
I moved it to `algebra/group_power/basic`. There was `add_pow_two` there (`(a + b) ^ 2 = a ^ 2 + 2 * a * b + b ^ 2`), as well as many related equalities/inequalities. I also added `sub_pow_two` for `(a-b)^2 = a ^ 2 - 2 * a * b + b ^ 2`.

Reply:
Ah, good idea!